Automattic/jetpack
# Classic Theme Helper: Modify Social Links package file to the version that will be used when required

## Proposed changes:

* This Pull request is a continuation of https://github.com/Automattic/jetpack/pull/38593, but modifying the new Classic Theme Helper package Social Links file based on issues discovered whilst requiring the package in this PR: https://github.com/Automattic/jetpack/pull/38730
* As an update to the info below which is not relevant anymore, after an internal review this file has mainly been reverted back to what it was - plus the non-striked-through changes below.
* ~This PR takes those changes added in PR 38730 and adds them independently (prior to requiring the package version), to prevent fatal errors in testing due to the vendor version not having the same functions.~
* The reason for the `social-links.php` (now `class-social-links.php`) changes, taken from PR 38730, are:
  * ~The `__construct` version has been replaced by `init` and `setup` functions, in alignment with `class-featured-content.php`.~
  * ~Functions are now static, also in alignment with the `class-featured-content.php` file, and tieing in with the creation of the `init` and `setup` functions - this means we can call the `setup` function directly.~
  * ~The last two functions in the file have been made public, so that they can be called by the Jetpack module file version if needed.~
  * Classes used (`Jetpack_Options`, `Publicize`, `WP_Customize_Manager` have been explicitly brought in via `use`.
* ~The reason those changes were made was because of issues calling / initializing the class via the Jetpack Mu WPcom package, and since the approach used by Featured Content works well it made sense to follow that approach.~

### Other information:


## Jetpack product discussion

See https://github.com/Automattic/jetpack/pull/38730
p1723190202324309-slack-C05PV073SG3

## Does this pull request change what data or activity we track or use?

No.

## Testing instructions:

* Technically the package isn't being used yet, and no build is happening, so a code check is the main thing.
* However originally in testing I did come across a fatal and had to make a few changes to make sure the package version didn't try to initialize the class as the file was being read, so we should make sure functionality is not affected. To do so:
  
Self-hosted and Atomic:
* On a test site, add a theme that includes support for Jetpack's social-links. The only theme I could find is 'Syntax'. 
* Otherwise, make sure you have a classic (not block) theme, such as Twenty Nineteen, and add the following snippet with a functionality plugin (like 'Code Snippets'):
```
add_action( 'after_setup_theme', function () {
	add_theme_support( 'social-links', array(
		'facebook', 'twitter', 'linkedin', 'tumblr',
	) );
} );
```
* Enable Social sharing - `/wp-admin/admin.php?page=jetpack#/sharing` and add a connection. 
* You can add a Facebook page by using the Gmail username and pw for `credsflow....` in the SS (internal only).
* Visit the Customizer, and you should see a 'Connect' menu item, that allows you to select your newly added connection.
* This should work both on trunk and with this PR applied.
* On Atomic make sure to test with the PR checked out for both Jetpack and WPcom Features, and each individually.
* You can also add an `error_log` in both the Jetpack module and Classic Theme Helper package files to confirm that it is the Jetpack file version that is loading (for Atomic I added to `jetpack-dev/jetpack-vendor/automattic/jetpack-classic-theme-helper...` as well as `jetpack-mu-wpcom-plugin/vendor/automattic/jetpack-classic-theme-helper...` to cover all bases.

Simple:
* Make sure you have a sandboxed Simple site and have applied both patches (as per the generated comment below).
* Add a social connection at `https://wordpress.com/marketing/connections/yoursite.com`
* Add a theme which includes support for Jetpack's social-links - Syntax for example.
* Visit the Customizer and you should see the 'Connect' menu item allowing you to connect your new social connection.
* This should work both with and without the patches applied - see testing comment below (and also one at a time to replicate Jetpack only and Mu Wpcom only).
* As above, you can also add an `error_log` to confirm the location of the file being used - tailing errors with `tail -f /tmp/php-errors`.
